Source/SeaInk.Core/Entities/Tables/GoogleTable.cs

```diff
@@ -0,0 +1,338 @@
+                    {
+                        Title = name
+                    },
+                    Fields = "title"
+                }
+            }.ToBody();
+
+            Service.Spreadsheets.BatchUpdate(requestBody, SpreadsheetId).Execute();
+        }
+
+        public void RenameSheet(TableIndex index, string name)
+        {
+            BatchUpdateSpreadsheetRequest requestBody = new Request
+            {
+                UpdateSheetProperties = new UpdateSheetPropertiesRequest
+                {
+                    Properties = new SheetProperties
+                    {
+                        SheetId = index.SheetId,
+                        Title = name
+                    },
+                    Fields = "title"
+                }
+            }.ToBody();
+
+            Service.Spreadsheets.BatchUpdate(requestBody, SpreadsheetId).Execute();
+            index.SheetName = name;
+        }
+
+        public T GetValueForCellAt<T>(TableIndex index)
+        {
+            return GetValuesForCellsAt<T>(new TableIndexRange(index))[0][0];
+        }
+
+        public string GetValueForCellAt(TableIndex index)
+        {
+            return GetValueForCellAt<string>(index);
+        }
+
+        public List<List<T>> GetValuesForCellsAt<T>(TableIndexRange range)
+        {
+            ValueRange result = Service.Spreadsheets.Values
+                .Get(SpreadsheetId, range.ToString())
+                .Execute();
+
+            if (result.Values.Count == 0)
+                throw new NonExistingIndexException();
+
+            return (List<List<T>>) Convert.ChangeType(
+                result.Values.Select(r => r.ToList()).ToList(),
+                typeof(List<List<T>>));
+        }
+
+        public List<List<string>> GetValuesForCellsAt(TableIndexRange range)
+        {
+            return GetValuesForCellsAt<string>(range);
+        }
+
+        public void SetValueForCellAt<T>(TableIndex index, T value)
+        {
+            SetValuesForCellsAt(index, new List<List<T>>
+            {
+                new List<T>
+                {
+                    value
+                }
+            });
+        }
+
+        public void SetValuesForCellsAt<T>(TableIndex index, List<List<T>> values)
+        {
+            var body = new ValueRange
+            {
+                Values = values.Select(v => v.Cast<object>().ToList())
+                    .Cast<IList<object>>()
+                    .ToList(),
+                MajorDimension = "ROWS"
+            };
+
+            var range = new TableIndexRange(
+                index,
+                index
+                    .WithRow(index.Row + values.Count - 1)
+                    .WithColumn(index.Column + values.Select(v => v.Count).Max() - 1));
+
+            SpreadsheetsResource.ValuesResource.UpdateRequest request = Service.Spreadsheets.Values.Update(
+                body,
+                SpreadsheetId,
+                range.ToString());
```

Comment:
Вроде видно, что это и реквест (из названия) и что метод, который его выдаёт называется Update. Скоратим)
```suggestion
            var request = Service.Spreadsheets.Values.Update(body, SpreadsheetId, range.ToString());
```